### PR TITLE
Clear error 40 before robot load

### PR DESCRIPTION
**Pull request by shihab-dls**
This should fix issue #505, where error 40 represents a common error of tripping the light curtains, and thus should be cleared before robot load.

Requires https://github.com/DiamondLightSource/mx-bluesky/pull/722

### CODE REVIEW

File: src/mx_bluesky/hyperion/experiment_plans/robot_load_and_change_energy.py [hunk 1]
```diff
@@ -122,6 +122,11 @@ def do_robot_load(
     demand_energy_ev: float | None,
     thawing_time: float,
 ):
+    error_code = yield from bps.rd(composite.robot.error_code)
+    # Reset robot if light curtains were tripped
```
Comment (olliesilvester) [inline src/mx_bluesky/hyperion/experiment_plans/robot_load_and_change_energy.py:126]: 😍

File: tests/unit_tests/hyperion/experiment_plans/test_robot_load_and_change_energy.py [hunk 2]
```diff
@@ -165,6 +167,40 @@ async def test_when_prepare_for_robot_load_called_then_moves_as_expected(
...
+    set_mock_value(robot_load_and_energy_change_composite.robot.error_code, 40)
+
+    RE = RunEngine()
+    RE(
+        do_robot_load(
+            robot_load_and_energy_change_composite, sample_location, demand_energy_ev, 0
+        )
+    )
+
+    mock_trigger.assert_called()
```
Comment (olliesilvester) [inline tests/unit_tests/hyperion/experiment_plans/test_robot_load_and_change_energy.py:201]: nit: Test _slightly_ easier to understand if this is called `mock_reset_errors`

**Review by olliesilvester — APPROVED** [review]
Nice one! One very very minor comment, take it or leave it

File: tests/unit_tests/hyperion/experiment_plans/test_robot_load_and_change_energy.py (hunk 2 above)
Comment (olliesilvester) [reply tests/unit_tests/hyperion/experiment_plans/test_robot_load_and_change_energy.py:201]: Actually, also, better to use `assert_called_once()` here

Comment (shihab-dls) [reply tests/unit_tests/hyperion/experiment_plans/test_robot_load_and_change_energy.py:201]: Yeah that's definitely better! Amending

**Pull request merged**